### DISCUSSION TITLE
OrjsonResponse confusion

### DISCUSSION THREAD

**Discussion by victoraugustolls** [fastapi/fastapi#7383] [Questions] [answered]
### First check

* [X] I added a very descriptive title to this issue.
* [X] I used the GitHub search to find a similar issue and didn't find it.
* [X] I searched the FastAPI documentation, with the integrated search.
* [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [X] I already read and followed all the tutorial in the docs and didn't find an answer.
* [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [X] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Description

Hi! First of all, thanks for the awesome project!

My problem here is that I'm trying to figure out the difference between the following two options:
- Returning ORJSONResponse with the content being a BaseModel that I used construct to skip validation (I'm assuming that this skips validation for FastAPI, might not be the case)
- Returning just a BaseModel with the following config:

```
def orjson_dumps(v, *, default):
    return orjson.dumps(v, default=default).decode()


def to_camel(string: str) -> str:
    parts = string.split("_")
    return parts.pop(0).lower() + "".join(word.capitalize() for word in parts)


class OrjsonModel(BaseModel):
    class Config:
        alias_generator = to_camel
        allow_mutation = False
        allow_population_by_field_name = True
        json_dumps = orjson_dumps
        json_loads = orjson.loads
```

One thing I noticed is that ORJSONResponse does not `decode()` the `orjson.dumps()` call, which with the model I need to.

What I'm expect to find out is the most optimal way for having docs of the response model without any kind of validation, and at the same time squish the most performance out of it that I can.

What I expected to happen is that by returning a OrjsonModel I could remove `.decode()`, and by using `.construct()` from `pydantic` I would skip FastAPI responde validation.

Are my assumptions correct? Is there a better way of achieving what I have in mind? Thanks!

**Comment by raphaelauv** [comment]
The best is
```python

class Reco(BaseModel):
    userId: str
    site: str


@router.post('/user', response_model=Reco)
async def user_reco(user: RecoInput):
        # something 
        return ORJSONResponse(content={"site": "toto", "userId": "toto"})

```

Like this you have the performance of ORJSONResponse and also the documentation in the swagger

**Comment by victoraugustolls** [comment]
Thanks for the answer? But this way fast api would validate the model with pydantic, right? Something that I want to skip.

**Comment by raphaelauv** [comment]
It skip validation , just try !!!! Remove one field in the ORJSONResponse but still present in the response_model.

**Comment by victoraugustolls** [comment]
Hi @raphaelauv thanks for the tip!! It indeed skipped validation. So the expected outcome of `ORJSONResponse` is to never validate?

**Comment by raphaelauv** [comment, marked as answer]
Read the doc -> https://fastapi.tiangolo.com/advanced/response-directly/

```
It won't do any data conversion with Pydantic models, it won't convert the contents to any type, etc.
```

You can close this issue

**Comment by victoraugustolls** [comment]
Wow, for some reason I was 100% sure of this, due to a pr that was opened for skipping validation for orjson. Oh well, thanks a lot @raphaelauv , must have misread / skipped this page on the docs.

**Comment by tiangolo** [comment]
Thanks for the help here @raphaelauv ! :clap: :bow: 

Thanks for reporting back and closing the issue @victoraugustolls  :+1:
